lib/srv/desktop/rdp/rdpclient/src/lib.rs

```diff
@@ -712,6 +778,42 @@ pub unsafe extern "C" fn handle_tdp_sd_delete_response(
     }
 }
 
+/// handle_tdp_sd_list_response handles a TDP Shared Directory List Response message.
+///
+/// # Safety
+///
+/// client_ptr MUST be a valid pointer.
+/// (validity defined by https://doc.rust-lang.org/nightly/core/primitive.pointer.html#method.as_ref-1)
+///
+/// res.fso_list MUST be a valid pointer
+/// (validity defined by the validity of data in https://doc.rust-lang.org/std/slice/fn.from_raw_parts_mut.html)
+///
+/// each res.fso_list[i].path MUST be a non-null pointer to a C-style null terminated string.
+///
+/// This function MUST NOT hang on to any of the pointers passed in to it after it returns.
+/// All passed data that needs to persist after this function MUST be copied into Rust-owned memory.
```

Comment:
Why do we have an advisory that on relates to the inner function itself on the function docstring?

Reply:
I think I see your point. The general idea was to add this advisory on all these functions at the FFI, in order to be very explicit about what's expected of them in terms of memory safety. Are you suggesting that this would more appropriately located at `SharedDirectoryListResponse::from`?

Reply:
Yep. My thought process is that notes on the FFI functions are advisories to the caller mostly. For impl notes about something I usually lean on placing it as a non docstring inside the body since the caller doesn't care.